### DISCUSSION TITLE
Enhancement(Infra): Move away from Poetry

### DISCUSSION THREAD

**Discussion by JacobCoffee** [litestar-org/litestar#2031] [Potential Issues]
### Summary

This issue serves as a launchpad to potentially swap from Poetry to another solution. We don't have to change or go anywhere, but this is just to get an idea for options, thoughts, and concerns.

Poetry causes many issues, its dependency resolution sometimes takes multiple minutes, they have more than once made questionable changes (https://github.com/python-poetry/poetry/pull/6297), dependencies leak outside of their intended target (https://discord.com/channels/919193495116337154/919193495690936353/1131650830060245094, https://github.com/python-poetry/poetry/issues/4401).

### Drawbacks and Impact

- New tool = new learning curve for everyone
- CI Changes

#### Where to?

- [PDM](https://pdm.fming.dev/latest/) 
- [Hatch](https://hatch.pypa.io/latest/)
- PDM + Hatch's build system, Hatchling
- [Rye](https://github.com/mitsuhiko/rye/) _(experimental)_
- [Huak](https://github.com/cnpryer/huak) _(experimental)_

#### Other

This doesn't really solve the "The Python packaging ecosystem is terrible overall" issue...

> **__NOTE__**: PDM does have a potential issue around [PEP582](https://peps.python.org/pep-0582/) being rejected.

## Funding
* If you would like to see an issue prioritized, make a pledge towards it!
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/litestar-org/litestar/issues/2025">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2025/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/2025/pledge.svg">
</picture>
</a>

**Comment by peterschutt** [comment]
I noticed recently that `hatch` has been adopted by [attrs](https://github.com/python-attrs/attrs/pull/1094) (they are a project I've always viewed as setting a good standard as far as project/build/testing configuration goes).

One thing hatch doesn't have is lock files: [ref](https://github.com/pypa/hatch/issues/749), [ref](https://github.com/pypa/hatch/issues/716). However, that's not really an end-user issue, and seems we could do something to work around that by using `pip-tools` to compile a hashed set of requirements for development environments.

Given that hatch is a PyPA project - it probably will be a bit more "boring" than other package managers out there - but sometimes boring is a good thing.

Just my 2c.

**Reply by cofin** [reply]
I recently tested PDM + hatchling and found that it was a really solid `poetry` replacement.  It had nearly drop-in feature parity with most of the poetry commands I use.

**Comment by Goldziher** [comment]
Why is this an issue? Sorry, this is not decided.

**Comment by provinzkraut** [comment]
I'm guessing the issue was opened to discuss and potentially track this. And I would agree that this is not decided, so how about we keep it open in the meantime?

**Comment by Goldziher** [comment]
you can switch it to a discussion, but its not an issue...
